docs/guides/route-calculation.md

```diff
@@ -0,0 +1,113 @@
+
+Most programming languages have libraries that implement these algorithms, or you can implement them yourself if you need more control over the behavior.
+
+!!! note "Important: A* Heuristic Considerations"
+
+    A\* requires a heuristic function (H-value) that estimates the remaining cost to reach the destination. This heuristic must **never overestimate** the actual cost, or A\* won't guarantee finding the optimal route.
+
+    In New Eden, creating an accurate heuristic is challenging because:
+
+    - Systems may appear closer in 3D space but require longer routes due to gate connections.
+    - "Pocket space" regions can create misleading distance calculations.
+
+    **Recommendation**: Set the heuristic value to zero for A\* in EVE route calculation. This effectively makes A\* behave like Dijkstra's algorithm, ensuring optimal routes.
+
+### Reading Static Data
+
+To build your route calculation system, you need data about system connections and security status. This information is available in the [Static Data](../services/static-data/index.md) export (SDE).
+
+#### System Connections
+
+The SDE contains stargate connection data in the `mapStargates` table. Each stargate has:
+
+- **System ID**: The system containing this stargate.
+- **Destination System ID**: The system this stargate connects to.
+
+#### System Security Status
+
+System security status is available in the `mapSolarSystems` table, which contains:
+
+- **System ID**: Unique identifier for the system.
+- **Security status**: Security status of the system.
+
+#### Building the Connection Graph
+
+To create a graph for pathfinding:
+
+1. **Load all systems** from `mapSolarSystems` to get security status.
+2. **Load all stargates** from `mapStargates` to get connections.
+3. **Create bidirectional edges** between connected systems.
+4. **Add connections** for Ansiblex (player-owned jump bridges), Wormholes, or Shipcasters.
+
+This graph can then be used with Dijkstra's algorithm or A* to find optimal routes based on your chosen cost function.
+
+### Cost function
+
+The cost function is the heart of your route calculation system. It determines how "expensive" each system transition is, influencing which route the algorithm will choose.
+EVE Online provides three distinct cost function modes, each with different security preferences.
+
+#### Shorter
+
+This mode finds the route with the fewest jumps, regardless of system security status.
+
+--8<-- "snippets/formulae/route-shorter.md"
+
+#### Safer
+
+This mode prioritizes routes through high-security space, making longer routes acceptable if they avoid dangerous systems.
+The lower the "security penalty" (0-100), the more likely the route will use low-security or null-security space.
+
+--8<-- "snippets/formulae/route-safer.md"
+
+#### Less-Secure
+
+This mode finds routes that prefers low-security space.
```

Comment:
TIL, I always thought it just prefers lower security numbers (so prefers null sec)